Assets/Scripts/Select Tool/EditablePlatformScript.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor.Experimental.GraphView;
+using UnityEngine;
+
+public class EditablePlatformScript : MonoBehaviour
+{
+    [SerializeField] private float maxXRel;
+    [SerializeField] private float minXRel;
+    [SerializeField] private float maxYRel;
+    [SerializeField] private float minYRel;
+    [SerializeField] private BoxCollider2D boxCollider;
+    [SerializeField] private LineRenderer lineRenderer;
+    
+    private float originalWidth;
+    private float originalHeight;
+    private float originalPosX;
+    private float originalPosY;
+    private Vector2 size;
+    
+
+    private void Start()
+    {
+        initValues();
+        lineRenderer.positionCount = 4;
+        lineRenderer.SetPosition(0, new Vector3(getMinX(), getMinY(), 0));
+        lineRenderer.SetPosition(1, new Vector3(getMinX(), getMaxY(), 0));
+        lineRenderer.SetPosition(2, new Vector3(getMaxX(), getMaxY(), 0));
+        lineRenderer.SetPosition(3, new Vector3(getMaxX(), getMinY(), 0));
+        disableBoundingBox();
+    }
+
+    public float getMaxX()
+    {
+        return originalPosX + size.x / 2f + maxXRel;
+    }
+
+    public float getMinX()
+    {
+        return originalPosX - size.x / 2f - minXRel;
+    }
+
+    public float getMaxY()
+    {
+        return originalPosY + size.y / 2f + maxYRel;
+    }
+
+    public float getMinY()
+    {
+        return originalPosY - size.y / 2f - minYRel;
+    }
+
+    private void OnDrawGizmos()
+    {
+        initValues();
+        // Green
+        Gizmos.color = new Color(0.0f, 1.0f, 0.0f);
```

Comment:
Can you utilize a game settings scriptable object instead to save these values? We will need to change them in the future and they seem like magic nums to me.

Reply:
These values are only visible in the inspector and not shown in game. I don't think it'd be appropriate to use a game settings object since this code doesn't run at runtime.